src/features/device-logs/lib/backends/loki.ts

```diff
@@ -210,11 +206,23 @@ export class LokiBackend implements DeviceLogsBackend {
 		const pushRequest = new PushRequest();
 		pushRequest.setStreamsList(streams);
 		const startAt = Date.now();
-		return this.pusher
-			.push(pushRequest, createOrgIdMetadata(String(appId)), {
-				deadline: Date.now() + PUSH_TIMEOUT,
-			})
-			.finally(() => updateLokiPushDurationHistogram(Date.now() - startAt));
+		return new Promise<any>((resolve, reject) => {
+			this.pusher.push(
+				pushRequest,
+				createOrgIdMetadata(String(appId)),
+				{
+					deadline: Date.now() + PUSH_TIMEOUT,
```

Comment:
It'd make sense to reuse the date we just created to avoid some work:
```suggestion
					deadline: startAt + PUSH_TIMEOUT,
```